### DISCUSSION TITLE
How to use a profile when publishing from code

### DISCUSSION THREAD

**Discussion by sam-wheat** [microsoft/DacFx#161] [Q&A]
I have variables I need to substitute during publish.  I use XML publish profile files.   I would like to use those files when publishing from code.  I believe the object I need to use is `DacProfile`.  I don't know how to use it in my code as shown below:

        var dacpac = DacPackage.Load("...");
        var dacpacService = new DacServices(connectionString);
        dacpacService.Publish(dacpac, "MyDB", new PublishOptions());

**Comment by ErikEJ** [comment]
@sam-wheat 

A DacProfile has "DeployOptions" 

Then use .Deploy (instead of publish) and pass the DeployOptions
